infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/KubernetesInternalRuntime.java

```diff
@@ -380,23 +370,24 @@ private void waitMachines(
     }
   }
 
-  private void finishAllStartupTracesAsFailures(Throwable reason) {
-    machineStartupTraces
-        .entrySet()
-        .removeIf(
-            e -> {
-              finishSpanAsFailure(e.getValue(), reason.getMessage());
-              return true;
-            });
-  }
-
   /**
    * Returns a function, the result of which the completable stage that performs servers checks and
    * start of servers probes.
    */
   private Function<Void, CompletionStage<Void>> checkServers(
       List<CompletableFuture<?>> toCancelFutures, KubernetesMachineImpl machine) {
+
+    // Need to get active span here to allow use in returned function
+    final Span activeSpan = tracerUtil.getActiveSpan();
+
     return ignored -> {
+      final Span tracingSpan =
```

Comment:
Why not building that span before lambda block? This would allow us not separating active span retrieval and building span.

Reply:
The way the machine boot chain works means that if the span is defined before the lambda block, the trace starts when the chain is created (i.e. the check servers span starts at the same time as the wait running async span). Doing it this way enables the check servers span to start immediately after wait running async is done.

I agree that this is pretty opaque, I'll add a comment explaining this.

Reply:
Makes sense. Thanks for the explanation

Reply:
I mean:
```java
private void buildCheckServerSpan(Span activeSpan, KubernetesMachineImpl machine) {
              return tracerUtil.buildSpan(
              "CheckServers",
              activeSpan,
              getContext().getIdentity().getWorkspaceId(),
              machine.getName());
}
```
and call here 
```java
final Span tracingSpan = buildCheckServerSpan(activeSpan, machine);
```
But it is up to you whether to do this or not - rather matter of taste since we don't have any code purity standards in Che

Reply:
Added comment explaining the reasoning.